src/Curl/Curl.php

```diff
@@ -45,40 +45,63 @@ class Curl
     public $id = null;
 
     public $error = false;
+    /** @var int  */
     public $errorCode = 0;
+    /** @var null|string */
     public $errorMessage = null;
 
+    /** @var bool  */
     public $curlError = false;
+    /** @var int  */
     public $curlErrorCode = 0;
+    /** @var null|string  */
     public $curlErrorMessage = null;
 
+    /** @var bool  */
     public $httpError = false;
+    /** @var int  */
     public $httpStatusCode = 0;
+    /** @var null|string  */
     public $httpErrorMessage = null;
 
+    /** @var null|string  */
     public $baseUrl = null;
+    /** @var null|string  */
     public $url = null;
+    /** @var null|string  */
     public $effectiveUrl = null;
     public $requestHeaders = null;
     public $responseHeaders = null;
     public $rawResponseHeaders = '';
     public $response = null;
+    /** @var null|string  */
     public $rawResponse = null;
 
+    /** @var null|callable  */
     public $beforeSendFunction = null;
+    /** @var null|callable  */
     public $downloadCompleteFunction = null;
+    /** @var null|callable  */
     public $successFunction = null;
+    /** @var null|callable  */
     public $errorFunction = null;
+    /** @var null|callable  */
     public $completeFunction = null;
 
+    /** @var array  */
     private $cookies = array();
+    /** @var array  */
     private $responseCookies = array();
+    /** @var array */
     private $headers = array();
+    /** @var array  */
     private $options = array();
 
     private $jsonDecoder = null;
+    /** @var string  */
     private $jsonPattern = '/^(?:application|text)\/(?:[a-z]+(?:[\.-][0-9a-z]+){0,}[\+\.]|x-)?json(?:-[a-z]+)?/i';
     private $xmlDecoder = null;
+    /** @var string  */
```

Comment:
Yes they are necessary for 3 reasons:

a) You should simply always document your code. How should someone who uses the lib know what data type he can when he uses it. And guys like me who write extension should also know what data type they're working with

b) Typehinting in IDE just does not work if you don't phpDoc your Code

c) Code generators generate code with the correct comments/types if you document your attributes